### PR TITLE
OpenApiProvider: Dictionary support

### PR DESCRIPTION
**Pull request by toburger**
Closes: #173 

The unit tests for the dictionary on the pet store sample are working, but I fear the schema of the pet store app is a bit "simple" compared to a production size schema.

I had no luck when running the provider on the following schema: https://api.tourism.testingmachine.eu/swagger/v1/swagger.json

```
error FS3033: The type provider 'SwaggerProvider.OpenApiClientTypeProvider' reported an error:
Cannot compile object 'ODHActivityPoi_PoiProperty_Item' based on unresolved reference '#/components/schemas/PoiProperty'
```

Did I something wrong in the implementation?

### CODE REVIEW

File: src/SwaggerProvider.DesignTime/v3/DefinitionCompiler.fs [hunk 1]
```diff
@@ -296,6 +296,12 @@ type DefinitionCompiler (schema:OpenApiDocument, provideNullable) as this =
             | _ when schemaObj.Reference <> null && not <| schemaObj.Reference.Id.EndsWith(tyName) ->
                 ns.ReleaseNameReservation tyName
                 compileByPath <| schemaObj.Reference.ReferenceV3
+            | _ when schemaObj.Type = "object" && schemaObj.AdditionalProperties <> null -> // Dictionary ->
+                let elSchema = schemaObj.AdditionalProperties
+                ProvidedTypeBuilder.MakeGenericType(typedefof<Map<string, obj>>, [
+                    typeof<string>
+                    compileBySchema ns tyName elSchema true ns.RegisterType false
```
Comment (toburger) [inline src/SwaggerProvider.DesignTime/v3/DefinitionCompiler.fs:303]: I have no idea what this is doing... 😅

**Comment by toburger** [comment]
Looks good to me! 👍
Despite the performance hiccups, but that's not related to this PR.

**Comment by sergey-tihon** [comment]
Released as [SwaggerProvider 0.12.0](https://www.nuget.org/packages/SwaggerProvider/0.12.0)
Thank you @toburger  for putting it all together!

**Comment by toburger** [comment]
Thanks for the help Sergey!

**Pull request merged**